File: Algorithm-CPP/18_층간소음.cpp
#include <iostream>
using namespace std;

bool calculateValue(int value, int max) {
    return value > max;
}

int main() {
    int n, m, cnt = 0, max = 0;
    cin >> n >> m;

    for(int i = 0; i < n; i++) {
        int value;
        cin >> value;

        if(calculateValue(value, m)) {
            cnt++;
            if(cnt > max) {
                max = cnt;
            }
        } else {
            cnt = 0;
        }
    }

    if(max == 0) cout << "-1" << endl;
    else cout << max << endl;

    return 0;
}

File: Algorithm-Swift/Algorithm-Swift/Inflearn/18_층간소음.swift
//
//  18_층간소음.swift
//  Algorithm-Swift
//
//  Created by SHIN YOON AH on 2023/07/31.
//

import Foundation

func solution18() {
    let nm = readLine()!.split(separator: " ").map { Int($0)! }
    let m = nm.last!
    let nums = readLine()!.split(separator: " ").map { Int($0)! }
    var cnt = 0, max = 0

    nums.forEach { num in
        if num > m {
            cnt += 1
            if cnt > max {
                max = cnt
            }
        } else {
            cnt = 0
        }
    }

    print(max)
}
